src/client/app/containers/ExportContainer.ts

```diff
@@ -0,0 +1,130 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { connect } from 'react-redux';
+import * as moment from 'moment';
+import * as _ from 'lodash';
+import ExportComponent from '../components/ExportComponent';
+import { chartTypes } from '../reducers/graph';
+import { ExportDataSet } from '../types/readings';
+import { State } from '../types/redux';
+
+/**
+ * @param {State} state
+ * @return {{meterInfo: *, selectedMeters: Array}}
+ */
+function mapStateToProps(state: State) {
+	const timeInterval = state.graph.timeInterval;
+	const datasets: ExportDataSet[] = [];
+	const chart = state.graph.chartToRender;
+	const barDuration = state.graph.barDuration;
+
+	if (chart === chartTypes.line) {
```

Comment:
The following code is horrifying and repetitive. Please see if you can simplify it by whatever means possible.